### PR TITLE
Only validate new users when they try to join


### PR DESCRIPTION
**Pull request by gabebw**
Most of our `before_action`s are about validating *new* users. But most of our traffic is *existing* users who already have a `Redirection` and we just need to follow it.

For those existing users, validating the slug, referrer, etc is a waste of resources. Now we skip doing that work.

### CODE REVIEW

File: app/controllers/redirections_controller.rb [hunk 1]
```diff
@@ -20,9 +20,8 @@ def previous
   private
```
Comment (gabebw) [inline app/controllers/redirections_controller.rb:20]: This diff is best viewed with whitespace ignored: https://github.com/hotline-webring/hotline-webring/pull/373/files?diff=unified&w=1

File: app/controllers/redirections_controller.rb [hunk 2]
```diff
@@ -73,33 +72,33 @@ def tagged_log(message)
...
+        redirect_to page_path("localhost")
+      end
     end
   end
 
   def ensure_referrer_is_not_blocked
-    if Redirection.exists?(slug: params[:slug])
-      # If the redirection already exists, let it through. We just want to
-      # prevent further redirections from being created.
-      return
```
Comment (gabebw) [inline app/controllers/redirections_controller.rb:85]: I replaced this block with `if creating_new_redirection?`, which does the same thing.

File: app/controllers/redirections_controller.rb [hunk 3]
```diff
@@ -73,33 +72,33 @@ def tagged_log(message)
   end
 
   def ensure_referrer_is_not_localhost
-    if HostValidator.new(referrer).invalid?
-      redirect_to page_path("localhost")
+    if creating_new_redirection?
```
Comment (edwardloveall) [inline app/controllers/redirections_controller.rb:75]: If you like you could also put these in the `before_action`s:

```rb
before_action :ensure_referrer_is_not_localhost, if :creating_new_redirection?
```

**Review by edwardloveall — APPROVED** [review]

File: app/controllers/redirections_controller.rb (hunk 3 above)
Comment (gabebw) [reply app/controllers/redirections_controller.rb:75]: Yeah, but:

1. I dislike the DSL for stuff that's already in Ruby
2. I want to move these to their own classes someday and it's a bit easier if everything is in the function already.

**Pull request merged**